hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackScatter.java

```diff
@@ -335,9 +335,22 @@ protected List<DatanodeDetails> chooseDatanodesInternal(
                       .flatMap(List::stream).collect(Collectors.toList()),
                 requiredReplicationFactor);
     if (!placementStatus.isPolicySatisfied()) {
-      String errorMsg = "ContainerPlacementPolicy not met. Misreplication" +
-              " Reason: " + placementStatus.misReplicatedReason();
-      throw new SCMException(errorMsg, null);
+      ContainerPlacementStatus initialPlacementStatus =
+              validateContainerPlacement(
+                      Stream.of(usedNodes)
+                      .flatMap(List::stream).collect(Collectors.toList()),
+                      requiredReplicationFactor);
+      if (initialPlacementStatus.misReplicationCount()
+              < placementStatus.misReplicationCount()) {
+        String errorMsg = "ContainerPlacementPolicy not met. Misreplication" +
+                " Reason: " + placementStatus.misReplicatedReason() +
+                " Initial Used nodes mis-replication Count: " +
+                initialPlacementStatus.misReplicationCount() +
+                " Used nodes + Chosen nodes mis-replication Count: " +
+                placementStatus.misReplicationCount();
+        throw new SCMException(errorMsg, null);
```

Comment:
Should this error have code "FAILED_TO_FIND_SUITABLE_NODE" instead of null? I guess we found a node, and it doesn't meet the policy, so I guess it is a failure to find a suitable node.

These codes are not really used much now, but we might be able to use them later.

Reply:
done

Reply:
Done